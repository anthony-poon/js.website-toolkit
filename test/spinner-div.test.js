import React from "react";
import renderer from 'react-test-renderer';
import {SpinnerDiv} from "website-toolkit"

test("Render Test", () => {
    const inDiv = renderer.create(
        <SpinnerDiv in={true}>Testing</SpinnerDiv>
    );
    let tree = inDiv.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    const outDiv = renderer.create(
        <SpinnerDiv in={false}>Testing</SpinnerDiv>
    );
    // re-rendering
    tree = outDiv.toJSON();
    expect(tree).toMatchSnapshot();
});
