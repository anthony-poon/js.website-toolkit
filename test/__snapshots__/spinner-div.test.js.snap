// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test 1`] = `
<div
  className="async-div__wrapper "
>
  Testing
</div>
`;

exports[`Render Test 2`] = `
<div
  className="async-div__wrapper "
>
  <div
    className="async-div__popover "
  >
    <span
      aria-hidden="true"
      className="spinner-border spinner-border-sm text-primary"
      role="status"
    />
  </div>
  Testing
</div>
`;
